packages/block-editor/src/components/inserter/index.js

```diff
@@ -83,7 +93,23 @@ class Inserter extends Component {
 	}
 
 	render() {
-		const { position } = this.props;
+		const { position, hasOneAllowedItem, createIfOne } = this.props;
+		let toggle = null;
+		if ( hasOneAllowedItem ) {
+			toggle = () => {
+				return (
+					<Button
+						className={ classnames( 'block-editor-button-block-appender' ) }
+						onClick={ createIfOne }
+					>
+						<span className="screen-reader-text">{ __( 'Add Block' ) }</span>
```

Comment:
We should try to use the name of the child block here.

Reply:
Good idea, will try

Reply:
Ok, there is a baked in way, but I am not too proud of it considering a split on a slash and a replace of hyphens is necessary.